components/statistics/portal_statistics.lua

```diff
@@ -0,0 +1,1413 @@
+						shortname = item.shortname
+					}
+				)
+			end
+			if placement == FIRST and item.liquipediatiertype == SHOWMATCH then
+				data[opponent].showWins = data[opponent].showWins + 1
+			elseif item.liquipediatiertype ~= SHOWMATCH then
+				data[opponent][placement] = data[opponent][placement] + 1
+			end
+		end
+	end
+
+	Lpdb.executeMassQuery('placement', queryParameters, processData)
+
+	return data
+end
+
+
+--[[
+Section: Display Functions
+]]--
+
+
+---@param args table
+---@param parameters table
+---@param tagType string
+---@return Html
+function StatisticsPortal._returnGameCell(args, parameters, tagType)
+	local gameCell = mw.html.create(tagType)
+	if Logic.readBool(args.multiGame) and not parameters.game then
+		gameCell:wikitext('Total')
+	elseif Logic.readBool(args.multiGame) then
+		gameCell:wikitext(parameters.game)
+	end
+	return gameCell
+end
+
+
+---@param args table
+---@return Html
+function StatisticsPortal._earningsTableHeader(args)
+	local columnText = args.opponentType == Opponent.team and 'Organization' or 'Player'
+
+	local row = mw.html.create('tr')
+		:tag('th'):wikitext('#'):addClass('unsortable'):done()
+		:tag('th'):wikitext(columnText):addClass('unsortable'):done()
+		:tag('th'):wikitext('Achievements'):css('width', '200px'):addClass('unsortable'):done()
+		:tag('th'):wikitext(Medal['1']):done()
+		:tag('th'):wikitext(Medal['2']):done()
+		:tag('th'):wikitext(Medal['3']):done()
+
+	if Logic.readBool(args.displayShowMatches) then
+		row:tag('th'):wikitext('Show<br>Match')
+	end
+
+	row:tag('th')
+		:tag('abbr')
+		:attr('title', 'Total earnings across all games')
+		:wikitext('Earnings')
+
+	return row
+end
+
+
+---@param args table
+---@param placements table
+---@param earnings number
+---@param opponentIndex number
+---@param opponentDisplay Html
+---@return Html
+function StatisticsPortal._earningsTableRow(args, placements, earnings, opponentIndex, opponentDisplay)
+	local row = mw.html.create('tr')
+		:css('line-height', '25px')
+		:css('text-align', 'center')
+		:tag('td'):wikitext(opponentIndex):done()
+		:tag('td'):css('text-align', 'left'):node(opponentDisplay):done()
+		:tag('td'):wikitext(StatisticsPortal._achievementsDisplay(placements.sWinData or {})):done()
+		:tag('td'):wikitext(placements['1'] or '0'):done()
+		:tag('td'):wikitext(placements['2'] or '0'):done()
+		:tag('td'):wikitext(placements['3'] or '0'):done()
+
+	if Logic.readBool(args.displayShowMatches) then
+		row:tag('td'):wikitext(placements.showWins or '0')
+	end
+
+	row:tag('td')
+		:css('text-align', 'right')
+		:wikitext(Currency.display(US_DOLLAR, earnings, CURRENCY_FORMAT_OPTIONS))
+
+	return row
+end
+
+
+---@param data table
+---@return string
+function StatisticsPortal._achievementsDisplay(data)
+	local output = ''
+	if data and type(data[1]) == 'table' then
+		for _, item in ipairs(data) do
+			item.icon = string.gsub(item.icon, 'File:', '')
+			item.iconDark = string.gsub(item.iconDark or '', 'File:', '')
```

Comment:
item.icon can never be null but iconDark can?

Reply:
so there should be `item.icon = string.gsub(item.icon or '', 'File:', '')` instead?